internal/consensus/reactor_test.go

```diff
@@ -131,14 +137,8 @@ func setup(t *testing.T, numNodes int, states []*State, size int) *reactorTestSu
 	rts.network.Start(t)
 
 	t.Cleanup(func() {
-		for nodeID, r := range rts.reactors {
-			require.NoError(t, rts.states[nodeID].eventBus.Stop())
-			require.NoError(t, r.Stop())
-			require.False(t, r.IsRunning())
-		}
-
-		leaktest.Check(t)
 		cancel()
+		leaktest.Check(t)
```

Comment:
I know we weren't before, but do we need to wait for shutdown to complete before we run the leak detector to avoid races on timing?

Reply:
Oh. And I think this returns a closure we need to invoke:
```go
leaktest.Check(t)()
```

Reply:
oh god I'm going to do this in a different PR